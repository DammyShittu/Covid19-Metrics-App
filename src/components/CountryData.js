import React from 'react';
import PropTypes from 'prop-types';
import { FaArrowCircleRight } from 'react-icons/fa';
import covid from '../images/covid.png';
import '../styling/countryData.css';

const CountryData = ({ name, confirmedCases }) => (
  <li className="country-data">
    <div className="right">
      <FaArrowCircleRight className="arrow" />
    </div>
    <div className="country">
      <img src={covid} alt="Covid19 Logo" className="covid-img" />
      <aside>
        <h2>{name}</h2>
        <span>Total Confirmed</span>
        <h3>{`${confirmedCases} cases`}</h3>
      </aside>
    </div>
  </li>
);

CountryData.propTypes = {
  name: PropTypes.string.isRequired,
  confirmedCases: PropTypes.number.isRequired,
};

export default CountryData;
