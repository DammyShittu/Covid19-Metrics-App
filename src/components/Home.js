import { React, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCountriesFromApi } from '../redux/Home/Home';

const Home = () => {
  const state = useSelector((state) => state.countriesReducer);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCountriesFromApi());
  }, []);

  return (
    <ul>
      {state && state.map((country) => {
        const { name, confirmedCases, id } = country;
        return (
          <li key={id}>
            <h2>{name}</h2>
            <h3>{confirmedCases}</h3>
          </li>
        );
      })}
    </ul>
  );
};

export default Home;
