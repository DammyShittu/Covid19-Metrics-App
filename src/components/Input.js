import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ filter, onChange }) => (
  <div>
    <input
      type="text"
      value={filter}
      onChange={onChange}
      placeholder="Search Country"
    />
  </div>
);

Input.propTypes = {
  filter: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Input;
