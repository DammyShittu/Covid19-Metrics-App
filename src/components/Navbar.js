import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { AiOutlineArrowLeft } from 'react-icons/ai';
import { FaMicrophone } from 'react-icons/fa';
import { MdSettings } from 'react-icons/md';

const Navbar = ({ text, year }) => (
  <section>
    <div>
      <Link to="/">
        <AiOutlineArrowLeft />
      </Link>
      <h1>{year}</h1>
    </div>
    <div>
      <h2>{text}</h2>
    </div>
    <div>
      <FaMicrophone />
      <MdSettings />
    </div>
  </section>
);

Navbar.propTypes = {
  text: PropTypes.string.isRequired,
  year: PropTypes.string,
};

Navbar.defaultProps = {
  year: '',
};

export default Navbar;
