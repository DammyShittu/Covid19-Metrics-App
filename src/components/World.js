import React from 'react';
import PropTypes from 'prop-types';
import world from '../images/world.png';
import '../styling/world.css';

const World = ({ totalConfirmed }) => (
  <section className="world-section">
    <div className="world">
      <div>
        <img src={world} className="world-img" alt="World Map" />
      </div>
      <aside>
        <h2>World</h2>
        <span>{`${totalConfirmed} cases`}</span>
      </aside>
    </div>
  </section>
);

World.propTypes = {
  totalConfirmed: PropTypes.number,
};

World.defaultProps = {
  totalConfirmed: 0,
};

export default World;
