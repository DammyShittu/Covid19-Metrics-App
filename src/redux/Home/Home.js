const FETCH_COUNTRIES = 'covid19-metrics-app/Home/FETCH_COUNTRIES';

export const fetchCountries = (payload) => ({
  type: FETCH_COUNTRIES,
  payload,
});

export const fetchCountriesFromApi = () => async (dispatch) => {
  const countries = await fetch('https://api.covid19tracking.narrativa.com/api/2021-10-10');
  const response = await countries.json();
  if (response) {
    const date = '2021-10-10';
    const regions = response.dates[date].countries;
    const countriesArr = Object.values(regions);
    const oneCountry = countriesArr.map((country) => (
      {
        name: country.name,
        confirmedCases: country.today_confirmed,
        id: country.id,
        allDeaths: country.today_deaths,
        recovered: country.today_recovered,
        openCases: country.today_open_cases,
        allRegions: country.regions,
      }
    ));
    dispatch(fetchCountries(oneCountry));
  }
};

// export const countriesThunk = async (dispatch) => {
//   const newCountries = await fetchCountriesFromApi();
//   console.log(newCountries, 'adedamola');
//   if (newCountries) {
//     dispatch(fetchCountries(newCountries));
//   }
// };

const initialState = [];

const countriesReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_COUNTRIES:
      return action.payload;
    default:
      return state;
  }
};

export default countriesReducer;
