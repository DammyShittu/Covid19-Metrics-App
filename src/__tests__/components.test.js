import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../components/Home';
import CountryData from '../components/CountryData';
import Input from '../components/Input';
import Navbar from '../components/Navbar';
import World from '../components/World';

describe('All components/pages render correctly', () => {
  test('Home component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Home />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CountryData component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <CountryData name="Belgium" confirmedCases={22} />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Input component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Input filter="den" onChange={() => {}} />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Navbar component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Navbar text="World" year="2021" />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('World component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <World totalConfirmed={456765} />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
