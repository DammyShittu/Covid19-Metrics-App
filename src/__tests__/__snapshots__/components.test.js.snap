// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All components/pages render correctly CountryData component renders correctly 1`] = `
<li
  className="country-data"
>
  <div
    className="right"
  >
    <svg
      className="arrow"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
      />
    </svg>
  </div>
  <div
    className="country"
  >
    <img
      alt="Covid19 Logo"
      className="covid-img"
      src="covid.png"
    />
    <aside>
      <h2>
        Belgium
      </h2>
      <span>
        Total Confirmed
      </span>
      <h3>
        22 cases
      </h3>
    </aside>
  </div>
</li>
`;

exports[`All components/pages render correctly Home component renders correctly 1`] = `
Array [
  <header>
    <nav
      className="nav"
    >
      <div
        className="year"
      >
        <a
          className="left-arrow"
          href="/"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            />
          </svg>
        </a>
        <h1>
          2021
        </h1>
      </div>
      <div>
        <span>
          World Covid-19 Stats
        </span>
      </div>
      <div
        className="settings"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          />
        </svg>
      </div>
    </nav>
  </header>,
  <section
    className="world-section"
  >
    <div
      className="world"
    >
      <div>
        <img
          alt="World Map"
          className="world-img"
          src="world.png"
        />
      </div>
      <aside>
        <h2>
          World
        </h2>
        <span>
          0 cases
        </span>
      </aside>
    </div>
  </section>,
  <section>
    <div
      className="search"
    >
      <input
        onChange={[Function]}
        placeholder="Search Country..."
        type="text"
        value=""
      />
    </div>
    <h3
      className="stats-name"
    >
      Stats By Country
    </h3>
    <ul
      className="container"
    />
  </section>,
]
`;

exports[`All components/pages render correctly Input component renders correctly 1`] = `
<div
  className="search"
>
  <input
    onChange={[Function]}
    placeholder="Search Country..."
    type="text"
    value="den"
  />
</div>
`;

exports[`All components/pages render correctly Navbar component renders correctly 1`] = `
<header>
  <nav
    className="nav"
  >
    <div
      className="year"
    >
      <a
        className="left-arrow"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          />
        </svg>
      </a>
      <h1>
        2021
      </h1>
    </div>
    <div>
      <span>
        World
      </span>
    </div>
    <div
      className="settings"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
        />
      </svg>
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
    </div>
  </nav>
</header>
`;

exports[`All components/pages render correctly World component renders correctly 1`] = `
<section
  className="world-section"
>
  <div
    className="world"
  >
    <div>
      <img
        alt="World Map"
        className="world-img"
        src="world.png"
      />
    </div>
    <aside>
      <h2>
        World
      </h2>
      <span>
        456765 cases
      </span>
    </aside>
  </div>
</section>
`;
